lib/auth/twitter_authenticator.rb

```diff
@@ -9,6 +9,16 @@ def enabled?
     SiteSetting.enable_twitter_logins
   end
 
+  def healthy?
+    connection =
+      Faraday.new(url: "https://api.twitter.com") do |config|
+        config.basic_auth(SiteSetting.twitter_consumer_key, SiteSetting.twitter_consumer_secret)
+      end
+    connection.post("/oauth2/token").status == 200
+  rescue Faraday::Error
```

Comment:
I wonder if we want to log the response with `Discourse.warn_exception` here for debugging? Not sure how detailed the response info will be

Reply:
I'm thinking when we back this with a database record we can include the datails there so we can potentially augment our messaging with it.